framework/OutStreams/PlotInterfaces/PlotInterface.py

```diff
@@ -0,0 +1,113 @@
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for specifying the input of cls.
+    """
+    spec = super().getInputSpecification()
+    return spec
+
+  def __init__(self):
+    """
+      Init of Base class
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'PlotInterface'
+    self.overwrite = True     # overwrite existing creations? # TODO not in input specs?
+    self.subDirectory = None  # directory to save generated files to # TODO should be in spec
+    self.filename = ''        # target file name? TODO should be in spec
+
+  def handleInput(self, spec):
+    """
+      Loads the input specs for this object.
+      @ In, spec, InputData.ParameterInput, input specifications
+      @ Out, None
+    """
+    super().handleInput(spec)
+
+  def initialize(self, inDict):
+    """
+      Function to initialize the OutStream. It basically looks for the "data"
+      object and links it to the system.
+      @ In, inDict, dict, contains all the Objects are going to be used in the
+      current step. The sources are searched into this.
+      @ Out, None
+    """
+    pass
+
+  def run(self):
+    """
+      Main run method.
+      @ In, None
+      @ Out, None
+    """
+    pass
+
+  ##################
+  # Utility
+  def findSource(self, name, stepEntities):
+    """
+      Find a source from the potential step sources.
+      @ In, name, str, name of the source
+      @ In, stepEntities, dict, entities from the Step
+      @ Out, findSource, object, discovered object or None
+    """
+    for out in stepEntities['Output']:
+      if out.name == name:
+        return out
+    for inp in stepEntities['Input']:
+      if inp.name == name:
+        return inp
+    for other in ['TargetEvaluation', 'SolutionExport']:
+      if other in stepEntities:
+        if stepEntities[other].name == name:
+          return stepEntities[other]
+    return None
+
+  def getInitParams(self):
+    """
+      This function is called from the base class to print some of the
+      information inside the class. Whatever is permanent in the class and not
+      inherited from the parent class should be mentioned here. The information
+      is passed back in the dictionary. No information about values that change
+      during the simulation are allowed.
+      @ In, None
+      @ Out, paramDict, dict, dictionary containing the parameter names as keys
+        and each parameter's initial value as the dictionary values
+    """
+    paramDict = {}
+    paramDict['Global Class Type                  '] = 'OutStreamManager'
+    paramDict['Specialized Class Type             '] = self.type
+    if self.overwrite:
+      paramDict['Overwrite output everytime called'] = 'True'
+    else:
+      paramDict['Overwrite output everytime called'] = 'False'
+    return paramDict
```

Comment:
Does RAVEN enforce newlines at the end of files? Ignore if not.

Reply:
I can't find that standard anywhere in our docs; does pep8 specify it?

Reply:
I found answers at https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file, I'll see if I can make my editor automate it.